### PR TITLE
Convert App component to a Functional component

### LINKED ISSUE #756: Convert App component to a Functional component
Currently the App component is a class based component and ideally should be a functional component using effects and no render function.

### PR DESCRIPTION
**Pull request by ciremusyoka**
Closes #756 
Refactor App component to function component and use react hooks.

### CODE REVIEW

File: src/App/App.tsx [hunk 1]
```diff
@@ -119,332 +119,317 @@ const APP_LOGIN_URL = BACKEND_ACTIVE ? BACKEND_LOGIN_URL : REACT_LOGIN_URL;
...
-        });
-        const dimensions = {
-          env: GA_ENV,
-          username,
-        };
-        setDimensions(dimensions);
-      }
+const App = () => {
+  useEffect(() => {
+    const user = (getUser(store.getState()) || {}).username || '';
```
Comment (moshthepitt) [inline src/App/App.tsx:124]: Should we rename this to "username"?

File: src/App/App.tsx [hunk 2]
```diff
@@ -119,332 +119,317 @@ const APP_LOGIN_URL = BACKEND_ACTIVE ? BACKEND_LOGIN_URL : REACT_LOGIN_URL;
...
+                  }
+                  return (
+                    <ConnectedOauthCallback
+                      SuccessfulLoginComponent={() => {
+                        return <Redirect to={HOME_URL} />;
+                      }}
+                      LoadingComponent={Loading}
                       providers={providers}
-                      authorizationGrantType={AuthGrantType}
+                      oAuthUserInfoGetter={oAuthUserInfoGetter}
```
Comment (moshthepitt) [inline src/App/App.tsx:398]: Why is this changing?

Comment (ciremusyoka) [reply src/App/App.tsx:398]: Git is taking it as a change but have looked at it and nothing changed. Initially that is were the `login component` was before the lines reduced. Now the `home component` has moved to that position and i think that is why it was registered as a change.

File: src/App/App.tsx (hunk 1 above)
Comment (ciremusyoka) [reply src/App/App.tsx]: variable renamed

**Comment by moshthepitt** [comment]
@ciremusyoka LGTM - please update the branch and then merge

**Pull request merged**